source/task_manager/crawl_tasks.py

```diff
@@ -2,116 +2,59 @@
-
-    environ = os.environ.copy()
-    environ['JAVA_HOME'] = '/usr/lib/jvm/java-7-oracle'
-    cca_dump_proc = subprocess.Popen([nutch_path, "commoncrawldump", "-outputDir", cca_dir,
-                                        "-segment", os.path.join(crawl.get_crawl_path(), 'segments')],
-                                        stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=environ)
-    stdout, stderr = cca_dump_proc.communicate()
-    sys.stderr.write(stdout)
-    sys.stderr.write(stderr)
-    crawl.save()
-    return "Dumping CCA data"
+from apps.crawl_space.viz.stream import NutchUrlTrails
 
+nutch_path = 'nutch'
+crawl_path = 'crawl'
+ache_path = 'ache'
+
+# TODO - provide Nutch Common Crawl dump when added to REST API
 
 class NutchTask(Task):
     abstract = True
 
-    def after_return(self, *args, **kwargs):
-        nutch_log_statistics(self.crawl)
-        self.crawl = Crawl.objects.get(pk=self.crawl.pk)
-        self.crawl.rounds_left -= 1
-        self.crawl.save()
-        if os.path.exists(os.path.join(self.crawl.get_crawl_path(), "stop")):
-            self.crawl.rounds_left = 0
-            self.crawl.save()
-            os.remove(os.path.join(self.crawl.get_crawl_path(), "stop"))
-            return
-        if self.crawl.rounds_left >= 1:
-            time.sleep(10)
-            nutch.delay(self.crawl)
-
-
 @shared_task(bind=True, base=NutchTask)
 def nutch(self, crawl, rounds=1, *args, **kwargs):
     self.crawl = crawl
-    call = [
-        crawl_path,
-        "--index",
-        "-D",
-        "elastic.index=%s" % self.crawl.index_name,
-        self.crawl.seeds_list.path,
-        self.crawl.get_crawl_path(),
-        "1",
-    ]
-    with open(os.path.join(crawl.get_crawl_path(), 'crawl_proc.log'), 'a') as stdout:
-        proc = subprocess.Popen(call, stdout=stdout, stderr=subprocess.PIPE,
-             preexec_fn=os.setsid)
-    # Check whether a CeleryTask already exists. If no, create the new object. If
-    # yes (IntegrityError), update the rows of the already existing object.
-    try:
-        self.crawl_task = CeleryTask(pid=proc.pid, crawl=self.crawl, uuid=self.request.id)
-        self.crawl_task.save()
-    except IntegrityError:
-        self.crawl_task = CeleryTask.objects.get(crawl=self.crawl)
-        self.crawl_task.pid = proc.pid
-        self.crawl_task.uuid = self.request.id
-        self.crawl_task.save()
-    stdout, stderr = proc.communicate()
-    # Because of the difference between the binaries in lib/nutch/bin/nutch and
-    # bin/nutch, the returncode is not always reliable. When nutch is called from
-    # bin/nutch the error codes do not work properly.
-    if proc.returncode > 0:
-        raise RuntimeError("Crawl has failed. Please review the crawl logs.")
-    return "Round Complete"
+    self.crawl_task = None
+    nutch_client = nutch_rest_api.Nutch()
+    seed_client = nutch_client.Seeds()
+
+    # TODO: Remove this Shim with Brittain's help
```

Comment:
@brittainhard - I need to chat with you about this.  I think we can get avoid writing/reading a local seeds list file by directly accessing the assigned seed list.